spinalcordtoolbox/utils/shell.py

```diff
@@ -39,14 +39,21 @@ def display_open(file, message="Done! To view results"):
 
 SUPPORTED_VIEWERS = ['fsleyes', 'fslview_deprecated', 'fslview', 'itk-snap', 'itksnap']
 
+IMTYPES_COLORMAP = {'anat': {'fsleyes': 'greyscale', 'fslview': 'Greyscale'},
```

Comment:
Now that this is right next to `SUPPORTED_VIEWERS`, I was wondering how the other 3 viewers are handled, so maybe this is a good place to add a comment.

In particular, the fact that the keys are not arbitrary strings (it's meaningful whether they contain the substring `seg`) is important to point out to whoever wants to add new image types in the future.

But, I do worry that this comment could become out-of-sync with the actual code later in the file. So, feel free to reject this suggestion.
```suggestion
# The 'fsleyes' colormaps are used for 'fsleyes'.
# The 'fslview' colormaps are used for 'fslview' and 'fslview_deprecated'.
# For 'itk-snap' and 'itksnap', imtypes which contain the substring 'seg' are colored as segmentations.
IMTYPES_COLORMAP = {'anat': {'fsleyes': 'greyscale', 'fslview': 'Greyscale'},
```

Reply:
We could always word it in such a way that it's a little more agnostic to the specific way we map imtypes to itksnap CLI options?

```suggestion
# The 'fsleyes' colormaps are used for 'fsleyes'.
# The 'fslview' colormaps are used for 'fslview' and 'fslview_deprecated'.
# 'itksnap', however, doesn't support specific colormaps, so it's excluded here
IMTYPES_COLORMAP = {'anat': {'fsleyes': 'greyscale', 'fslview': 'Greyscale'},
```

I think this comment would hold up even after most code changes?

Reply:
But for adding future image types to the map, it's important to point out that names matter for itksnap colors, not just that itksnap is excluded.

Reply:
Sure, that could work. For itksnap, the "colormap" values could be `'seg'` and `''`, for example, and handled by the code in `_construct_itksnap_syntax`.

Reply:
In the process of manually matching up ITKsnap commands with different image types, I discovered the following:

1. Softsegs, when combined with the `-s` option (seg), won't display at all!! 
   - There's not even an error -- the image is just missing entirely from the ITKSnap window. 
   - I had to use the normal grayscale option just to even get it to display. 
 ![image](https://user-images.githubusercontent.com/16181459/223781507-3ad5cb19-62dc-4db4-ac00-69f421ad1293.png)   
2. Label files, when combined with the `-s` option (seg), will display colorful labels.
  ![image](https://user-images.githubusercontent.com/16181459/223781230-c0214b3c-26f8-4e46-9bbb-92be1a075ad0.png)

I presume that the `-s` seg option checks for either 1 nonzero value (color=red) or >1 nonzero values (color=rainbow). But, if an image has too many nonzero values (in the case of softsegs), then not even the rainbow colormap is sufficient, and thus ITKSnap expects it to be treated as a normal greyscale image.

Both of these results went against my intuitions (softseg == no seg, labels == seg???), and _definitely_ suggest that we should be using 1:1 mappings between imtypes and ITKSnap options.

Reply:
Aside: I also found that `-s` (seg images) can only be used once per ITKsnap command. To account for this new information, I reworked the ITKSnap command so that it produces one command per seg file:

```python
>>> cmd = "itksnap -g t2.nii.gz"
>>> seg_images = ["t2_seg.nii.gz"]
>>> cmd = "\n".join([f"{cmd} -s {seg_image}" for seg_image in seg_images])
>>> print(cmd)
itksnap -g t2.nii.gz -s t2_seg.nii.gz

>>> cmd = "itksnap -g t2.nii.gz"
>>> seg_images = ["t2_seg_1.nii.gz", "t2_seg_2.nii.gz"]
>>> cmd = "\n".join([f"{cmd} -s {seg_image}" for seg_image in seg_images])
>>> print(cmd)
itksnap -g t2.nii.gz -s t2_seg_1.nii.gz
itksnap -g t2.nii.gz -s t2_seg_2.nii.gz

```

Making this change perhaps goes beyond the scope of this PR? But, a rewrite of the ITKSnap function was already needed to account for this new `imtype` behavior, so I don't think I'm doing _too_ much extra here...